master/buildbot/reporters/http.py

```diff
@@ -41,11 +41,28 @@ def checkConfig(self, *args, **kwargs):
                          (self.__class__,))
         if not isinstance(kwargs.get('builders'), (type(None), list)):
             config.error("builders must be a list or None")
+        # check that our callbacks are functions
+        if kwargs.get('startCB') and not hasattr(kwargs['startCB'], '__call__'):
+            config.error("start callback must be callable")
+        if kwargs.get('reviewCB') and not hasattr(kwargs['reviewCB'], '__call__'):
+            config.error("review callback must be callable")
+        if kwargs.get('summaryCB') and not hasattr(kwargs['summaryCB'], '__call__'):
+            config.error("summary callback must be callable")
 
     @defer.inlineCallbacks
-    def reconfigService(self, builders=None, **kwargs):
+    def reconfigService(self, builders=None,
+                        startCB=None, startArg=None,
+                        reviewCB=None, reviewArg=None,
+                        summaryCB=None, summaryArg=None,
+                        **kwargs):
         yield service.BuildbotService.reconfigService(self)
         self.builders = builders
+        self.startCB = startCB
+        self.startArg = startArg
+        self.reviewCB = reviewCB
```

Comment:
I don't think we should add more callbacks in the base class. 

* reviewCB does not make sense for most reporters. It is a gerrit thing

* I think it is much more flexible to have people subclass the reporters if they really want to change the details of the messages.

* perhaps we can put an hint for the event kind in send, rather than putting the callback